File: labs-code/r/academic-research-product-track/user_lookup.R
library(academictwitteR)

# Replace with your own bearer token
bearer_token <- "XXXXX"

# Replace with user IDs of your choice
user_ids <- c("2244994945", "6253282")

users <-
  get_user_profile(
    user_ids,
    bearer_token)

View(users)

File: labs-code/python/academic-research-product-track/user_lookup.py
from twarc import Twarc2, expansions
import json

# Replace your bearer token below
client = Twarc2(bearer_token="XXXXX")


def main():
    # List of user IDs to lookup, add the ones you would like to lookup
    users = ['2244994945', '783214', '6253282']
    # The user_lookup function gets the hydrated user information for specified users
    lookup = client.user_lookup(users=users)
    for page in lookup:
        result = expansions.flatten(page)
        for user in result:
            # Here we are printing the full Tweet object JSON to the console
            print(json.dumps(user))


if __name__ == "__main__":
    main()
